Author: Andrew Harle <46223597+andrewharle@users.noreply.github.com>
Description: Use MMAPv1 by default on i386 and ARM 32 bit.
 Since wiredTiger does not build on i386, make sure mongod won't try to use it
 by default.
Last-Update: 2019-01-03
Forwarded: no
--- a/src/mongo/db/storage/storage_options.h
+++ b/src/mongo/db/storage/storage_options.h
@@ -51,7 +51,11 @@
     static const char* kDefaultConfigDbPath;
 
     StorageGlobalParams()
+#if defined(__i386__) || defined(__arm__)
+        : engine("mmapv1"),
+#else
         : engine("wiredTiger"),
+#endif
           engineSetByUser(false),
           dbpath(kDefaultDbPath),
           upgrade(false),
--- a/src/mongo/dbtests/framework_options.cpp
+++ b/src/mongo/dbtests/framework_options.cpp
@@ -92,7 +92,11 @@
 
     options->addOptionChaining(
                  "storage.engine", "storageEngine", moe::String, "what storage engine to use")
+#if defined(__i386__) || defined(__arm__)
+        .setDefault(moe::Value(std::string("mmapv1")));
+#else
         .setDefault(moe::Value(std::string("wiredTiger")));
+#endif
 
     options->addOptionChaining("suites", "suites", moe::StringVector, "test suites to run")
         .hidden()
